key-manager.js

```diff
@@ -222,10 +223,12 @@ class KeyManager {
 
 module.exports = KeyManager
 
-/** @param {string} projectId */
-function nonceFromProjectId (projectId) {
-  return Buffer.from(projectId, 'hex').subarray(
-    0,
-    sodium.crypto_aead_xchacha20poly1305_ietf_NPUBBYTES
-  )
+/** @param {string} projectId z-base-32 encoded hash of the project key */
+function nonceFromProjectId(projectId) {
```

Comment:
looking at this, I'm wondering if this should be more flexible as opposed to assuming the usage of the `projectKeyToPublicId` util. Like what if the consumer creates a project id using some other means?